android-components/components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt

```diff
@@ -116,6 +118,7 @@ interface AwesomeBar {
         val onSuggestionClicked: (() -> Unit)? = null,
         val onChipClicked: ((Chip) -> Unit)? = null,
         val score: Int = 0,
+        val metadata: Map<String, Any>? = null,
```

Comment:
I couldn't think of a simpler way for `AwesomeBar.Suggestion` to pass data from a-c's `FxSuggestSuggestionProvider` to Fenix's `MetricsMiddleware`, and took some inspiration from `Fact`.

Reply:
Yeah, this works. If you really wanted to use some types here, something like this could work too:

```kotlin
interface MetadataItem

sealed interface FxSuggestInteractionInfo: MetadataItem { }

data class Suggestion(
  val provider: SuggestionProvider,
  // ...
  val metadata: Map<String, MetadataItem>? = null,
)
```

I'm fine with it as is since there is good prior art examples. If we need to make something that defines explicit serializable APIs, we could also do that in a follow-up.